src/vse_sync_pp/analyzers/ppsdpll.py

```diff
@@ -0,0 +1,209 @@
+### SPDX-License-Identifier: GPL-2.0-or-later
+
+import pandas as pd
+import numpy as np
+import allantools
+
+from collections import namedtuple
+from decimal import Decimal
+
+class Analyzer():
+    def __init__(self):
+        self._col1_name = 'timestamp'
+        self._col2_name = 'eecstate'
+        self._col3_name = 'phasestate'
+        self._col4_name = 'phaseoffset'
+        self._data  = pd.DataFrame(columns=['timestamp','eecstate','phasestate','phaseoffset']) 
```

Comment:
I would also enforce that colN_name vars and the data frame is consistant. 
```
self._data = pd.DataFrame(columns=[self._col1_name,self._col2_name, self._col3_name,self._col4_name])
```
Or have a mapping between them in a dict or something like that.